### DISCUSSION TITLE
Support for packaging and submitting loose files

### DISCUSSION THREAD

**Discussion by heftymouse** [microsoft/msstore-cli#8] [Ideas]
Currently, only a small number of supported frameworks can effectively be used with the Store CLI. Adding the ability to package from any directory and manifest file would allow the app to be used with any framework even without explicit support.

**Comment by azchohfi** [comment]
Thanks for the suggestion! 😀

I'm trying to understand the scenario a little bit better.

I can see two 3 different scenarios here. One is building an `msix` package with a loose set of files, second one is the same as the first, but in a non-Windows environment, and the third is publishing an MSIX package in non-Windows devices. The 3rd one I can see happening quite easily in terms of implementation, and we should definitely support. The first one is more complex, and I'm not sure why MakeAppx itself doesn't fulfill that need, and with the 2nd one I'm struggling to see the point in implementing it.

The `package` command is already very limited to work only on Windows for most of the project types that we support (exception is PWAs, since it is a service call for PWABuilder that can happen anywhere). These are limitations of the underlying build processes of the project types that we currently support. For example, Flutter projects can be developed on a MacOS device, but if you need to build for Windows (using the MSIX extension), it needs to happen on a Windows machine. You can still call the `init` command on a MacOS or Linux device and then run the `package` and `publish` commands on a Windows hosted build (e.g. in a hosted Windows GitHub Actions build pipeline).

I'm trying to think of a scenario/platform/framework where a cross-platform support for a MakeAppx equivalent would make sense. Does it make sense to build an exe on MacOS or Linux device and then package it as an MSIX right there, in the same build machine? Which platforms/framework does that?

Now, for the 3rd scenario of simply pointing to an MSIX and publishing it, I believe we should add that support. You could have a build that runs on Windows, generates an MSIX, and upload it as a build artifact, and in a different job (probably running on Linux), you download that artifact and upload it to the store, in a specific flight, etc.

**Reply by heftymouse** [reply]
Certain frameworks like Electron allow you to target Windows from Linux. However I agree that it makes little to have packaging be cross-platform. I'll remove that part from the original post.
